src/AspNetCoreModuleV2/AspNetCore/applicationinfo.cpp

```diff
@@ -66,48 +69,70 @@ APPLICATION_INFO::CreateHandler(
 HRESULT
 APPLICATION_INFO::CreateApplication(const IHttpApplication& pHttpApplication)
 {
-    HRESULT hr = S_OK;
-
     if (AppOfflineApplication::ShouldBeStarted(pHttpApplication))
     {
         LOG_INFO(L"Detected app_offline file, creating polling application");
-        #pragma warning( push )
-        #pragma warning ( disable : 26409 ) // Disable "Avoid using new", using custom deleter here
-        m_pApplication.reset(new AppOfflineApplication(pHttpApplication));
-        #pragma warning( pop )
+        m_pApplication.reset(std::make_unique<AppOfflineApplication>(pHttpApplication).release());
```

Comment:
I don't like this pattern

Reply:
Me neither but there are better ways of fixing it:

```

template< class APPLICATION, class PTR_TYPE >
std::unique_ptr<PTR_TYPE, IAPPLICATION_DELETER> make_application()
{
// put pragmas here.
    return new std::unique_ptr<PTR_TYPE, IAPPLICATION_DELETER>(new APPLICATION());
}

```

Reply:
Wouldn't you need a unique one for each application we create? Every subclass would need to implement this.

Reply:
No, that's why it has type arguments:
```
template< class APPLICATION, class PTR_TYPE, typename ...Params >
std::unique_ptr<PTR_TYPE, IAPPLICATION_DELETER> make_application(Params&&... params)
{
#pragma warning( push )
#pragma warning ( disable : 26409 ) // Disable "Avoid using new", using custom deleter here
    return std::unique_ptr<PTR_TYPE, IAPPLICATION_DELETER>(new APPLICATION(std::forward<Params>(params)...));
#pragma warning( pop )
}


Usage:
m_pApplication = make_application<AppOfflineApplication, IAPPLICATION>(pHttpApplication);

```